tools/regression_test.py

```diff
@@ -0,0 +1,772 @@
+
+        # update the report
+        update_report(
+            report_dict=report_dict,
+            model_name=model_cfg_path.parent.name,
+            model_config=str(model_cfg_path),
+            model_checkpoint_name=str(checkpoint_path),
+            dataset='',
+            backend_name=backend_name,
+            deploy_config=str(deploy_cfg_path),
+            static_or_dynamic=infer_type,
+            conversion_result=str(convert_result),
+            fps=fps,
+            metric_info=metric_list,
+            test_pass=str(test_pass))
+
+
+def save_report(report_info, report_save_path, logger):
+    """Convert model to onnx and then get metric.
+
+    Args:
+        report_info (dict):  Report info dict.
+        report_save_path (Path): Report save path.
+        logger (logging.Logger): Logger.
+    """
+    logger.info(f'Save regression test report '
+                f'to {report_save_path}, pls wait...')
+    try:
+        df = pd.DataFrame(report_info)
+        df.to_excel(report_save_path)
+    except ValueError:
+        logger.info(f'Got error report_info = {report_info}')
+
+    logger.info(f'Saved regression test report to {report_save_path}.')
+
+
+def main():
+    args = parse_args()
+
+    set_start_method('spawn')
+
+    logger = get_root_logger(log_level=args.log_level)
+    logger.info('Processing regression test.')
+
+    backend_file_info = {
+        'onnxruntime': 'end2end.onnx',
+        'tensorrt': 'end2end.engine',
+        'torchscript': 'end2end.pt',
+
+        # unknown
+        'openvino': '',
+        'ncnn': '',
+        'pplnn': '',
+    }
+
+    backend_list = args.backends
+    if backend_list == ['all']:
+        backend_list = [
+            'onnxruntime', 'tensorrt', 'openvino', 'ncnn',
+            'pplnn', 'torchscript'
+        ]
+    assert isinstance(backend_list, list)
+    logger.info(f'Regression test backend list = {backend_list}')
+
+    work_dir = Path(args.work_dir)
+    work_dir.mkdir(parents=True, exist_ok=True)
+
+    log_path = work_dir.joinpath('regression_test.log').absolute()
+    if log_path.exists():
+        with open(log_path, 'w') as f_log:
+            f_log.write('')  # clear the log file
+    logger.info(f'Savine log in {str(log_path)}')
+
+    for deploy_yaml in args.deploy_yml:
+
+        if not Path(deploy_yaml).exists():
+            raise FileNotFoundError(f'deploy_yaml {deploy_yaml} not found, '
+                                    'please check !')
+
+        with open(deploy_yaml) as f:
+            yaml_info = yaml.load(f, Loader=yaml.FullLoader)
+
+        report_save_path = \
+            work_dir.joinpath(Path(deploy_yaml).stem + '_report.xlsx')
+        report_dict = {
+            'model_name': [],
+            'model_config': [],
+            'model_checkpoint_name': [],
+            'dataset': [],
+            'backend_name': [],
+            'deploy_config': [],
+            'static_or_dynamic': [],
+            'conversion_result': [],
```

Comment:
we lost a column of precission type such as FP32, FP16, int8

Reply:
Done